src/pip/_internal/commands/install.py

```diff
@@ -439,14 +433,15 @@ def run(self, options, args):
 
             return ERROR
 
-        if options.target_dir:
+        if options.target_dir and target_temp_dir:
             self._handle_target_dir(
```

Comment:
Since `options.target_dir` implies `target_temp_dir` is set (see above in the same method), I'd suggest using an `assert target_temp_dir` here.

Reply:
Fixed in https://github.com/pypa/pip/pull/8365/commits/87249b9850bb390750f5e8e349c7ab6ccc75e359